src/components/switch/switch.stories.tsx

```diff
@@ -0,0 +1,64 @@
+import { useArgs } from '@storybook/preview-api';
+import type { Meta, StoryObj } from '@storybook/react';
+import { useState } from 'react';
+import { Switch } from './switch';
+
+const meta: Meta<typeof Switch> = {
+    title: 'components/Switch',
+    component: Switch,
+    tags: ['autodocs'],
+    parameters: {
+        design: {
+            type: 'figma',
+            url: 'https://www.figma.com/file/ISSDryshtEpB7SUSdNqAcw/branch/jfKRr1V9evJUp1uBeyP3Zz/Aragon-ODS?node-id=8850%3A12962&mode=dev',
+        },
+    },
+};
+
+type Story = StoryObj<typeof Switch>;
+
+/**
+ * `Switch` used as an uncontrolled component
+ */
+export const Uncontrolled: Story = {
+    args: {
+        label: 'Show testnets',
+        name: 'testnet',
+        defaultChecked: true,
+        onCheckedChanged: undefined,
+    },
+};
+
+/**
+ * Controlled usage of the `Switch` component
+ */
+export const Controlled: Story = {
+    decorators: [
+        (Story) => {
+            const [checked, setChecked] = useState(false);
+
+            const handleCheckedChanged = (newValue: boolean) => {
+                setChecked(newValue);
+            };
+
+            return <Story checked={checked} onCheckedChanged={handleCheckedChanged} />;
+        },
+    ],
+
+    args: {
+        label: 'Show testnets',
+        name: 'testnet',
+    },
+
+    render: function Component(args) {
+        const [, setArgs] = useArgs();
```

Comment:
On other stories we implement a component and pass it to the `render` function when the component needs to use hooks (e.g.  useState). Can't we do the same here? Is using `decorators` and `useArgs` better that using a component?

Reply:
Oh yeah, Storybook hasn't at all been playing nice with me when it comes to controlled components. When I render the function the usual way, the state doesn't get updated in the documentation page even though it works just fine for the individual story.

Reply:
Mm.. could you try something like that:
```
/**
 * Controlled usage of the `Switch` component
 */
const ControlledComponent = (props: ISwitchProps) => {
    const [checked, setChecked] = useState(false);

    return <Switch checked={checked} onCheckedChanged={setChecked} {...props} />;
};

export const Controlled: Story = {
    render: ({ onCheckedChanged, ...props }: ISwitchProps) => <ControlledComponent {...props} />,
    args: {
        label: 'Show testnets',
        name: 'testnet',
    },
};
```

It might be caused by `Storybook` setting a default function for the `onCheckedChanged` property

Reply:
This works halfway. It's not able to capture the action since `onCheckedChanged` is deliberately removed.

Reply:
Yes because we always set it on the `ControlledComponent` component through `onCheckedChanged={setChecked}` and prevent it to be overriden by the Storybook props. The above works on my end, let me know if you have issues and we can have a quick sync